### PR TITLE
BFD-1589 Refresh CODEOWNERS

### PR DESCRIPTION
**Pull request by mjburling**
**JIRA Ticket:**
[BFD-1589](https://jira.cms.gov/browse/BFD-1589)

**User Story or Bug Summary:**

Extend CODEOWNERS
---

### What Does This PR Do?

After a recent discussion on CODEOWNERS, we realized we needed the following corrections:
* add GitHub Actions Workflows section for /.github/workflows paths
* add @mjburling to mirror @keithdadkins in CODEOWNERS document

After making the above changes, I noticed that github complained about CODEOWNERS that do not have access to the repository. To tamp down on the noise, I'm invoking the campsite rule and removing those users, namely:
- `@msnook`
- `@orenfromberg`
- `@lailasharshar`
- `@wnyffenegger`

### What Should Reviewers Watch For?
* To see the current validation errors, view [CODEOWNERS](https://github.com/CMSgov/beneficiary-fhir-data/blob/master/.github/CODEOWNERS) in the trunk directly
* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

**Review by christrotteradhoc — APPROVED** [review]

**Comment by mjburling** [comment]
I need to tidy up a few things with this branch before proceeding in addition to some GitHub warnings.

**Review by lsmitchell — COMMENTED** [review]
Looks good; would just want a second look by Trotter before I approve (i'm not sure what goes into decision around who gets added to codeowners, if there are any)

File: .github/CODEOWNERS
Comment (keithdadkins) [inline .github/CODEOWNERS:8]: Might as well add yourself here too.

File: .github/CODEOWNERS
Comment (keithdadkins) [inline .github/CODEOWNERS:16]: could you add Trotter here?

Comment (keithdadkins) [reply .github/CODEOWNERS:16]: also to the other insights

**Review by keithdadkins — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS
Comment (mjburling) [reply .github/CODEOWNERS]: You bet. I think he's most everywhere now!

File: .github/CODEOWNERS
Comment (mjburling) [reply .github/CODEOWNERS]: Good idea. Thanks!

**Review by christrotteradhoc — APPROVED** [review]

**Review by keithdadkins — APPROVED** [review]
Thanks!

**Pull request merged**
